exporter/awsemfexporter/bounded_pusher_map.go

```diff
@@ -0,0 +1,76 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awsemfexporter
+
+import (
+	"errors"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/cwlogs"
+)
+
+const (
+	pusherMapLimit = 1000
+)
+
+type BoundedPusherMap struct {
```

Comment:
this utility seems out of place.  

My first thought is can we re-use something that already exists in this code-base? I see some code here using the popular hashicorp golang-lru as a cache

https://github.com/amazon-contributing/opentelemetry-collector-contrib/blob/b0f0c0eda01baee4d29f71621b06b80e082caeae/extension/ackextension/inmemory.go#L10

Reply:
There is also the "simple" version.  not sure what the difference is between simple and v2
https://github.com/amazon-contributing/opentelemetry-collector-contrib/blob/b0f0c0eda01baee4d29f71621b06b80e082caeae/connector/spanmetricsconnector/internal/cache/cache.go#L18